src/Utopia/Messaging/Adapters/SMS/GEOSMS.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Utopia\Messaging\Adapters\SMS;
+
+use Utopia\Messaging\Adapters\SMS as SMSAdapter;
+use Utopia\Messaging\Adapters\SMS\GEOSMS\CallingCode;
+use Utopia\Messaging\Messages\SMS;
+
+class GEOSMS extends SMSAdapter
+{
+    protected $defaultAdapter;
+
+    protected $localAdapters = [];
+
+    public function __construct(SMSAdapter $defaultAdapter)
+    {
+        $this->defaultAdapter = $defaultAdapter;
+    }
+
+    public function getName(): string
+    {
+        return 'GEOSMS';
+    }
+
+    public function getMaxMessagesPerRequest(): int
+    {
+        return PHP_INT_MAX;
+    }
+
+    public function setLocal(string $callingCode, SMSAdapter $adapter): self
+    {
+        $this->localAdapters[$callingCode] = $adapter;
+
+        return $this;
+    }
+
+    protected function process(SMS $message): string
+    {
+        $recipientsByCallingCode = $this->groupRecipientsByCallingCode($message->getTo());
+        $responses = [];
+
+        foreach ($recipientsByCallingCode as $callingCode => $recipients) {
+            $adapter = isset($this->localAdapters[$callingCode])
+                ? $this->localAdapters[$callingCode]
+                : $this->defaultAdapter;
+
+            $responses[] = $adapter->send(new SMS(
+                to: $recipients,
+                content: $message->getContent(),
+                from: $message->getFrom(),
+                attachments: $message->getAttachments()
+            ));
+        }
```

Comment:
@loks0n we might wanna think about adding some sort of error handling here as failure to send to one adapter will cause the others to fail as well